test/test_tensor.h

```diff
@@ -25,36 +25,22 @@ TEST(tensor, constructors) {
 
   // check that t2 values has been copyied to t1
   for (size_t i = 0; i < t1.size(); ++i) {
-    EXPECT_EQ(t1.host_data()[i], float_t(2.0));
+    EXPECT_EQ(*std::next(t1.host_begin(), i), float_t(2.0));
```

Comment:
Actually this is a dirty fix that somehow made it to final version. Will think about something more elegant